lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -655,6 +697,7 @@ class LokiPushApiEndpointJoined(EventBase):
 class LoggingEvents(ObjectEvents):
     """Event descriptor for events raised by `LokiPushApiProvider`."""
 
+    loki_push_api_alert_rules_error = EventSource(LokiPushApiAlertRulesError)
```

Comment:
`loki_push_api_alert_rules_error` is well scoped, and is an instance of the well named `LokiPushApiAlertRulesError`.
I wonder if we should drop the prefix here
```suggestion
    alert_rules_error = EventSource(LokiPushApiAlertRulesError)
```
so the user could just
```python
        self.framework.observe(
            self._loki_consumer.on.alert_rules_error,
            self._alert_rules_error
        )
```

Reply:
If we remove the prefix here, we should also remove:

```python
class LoggingEvents(ObjectEvents):
    """Event descriptor for events raised by `LokiPushApiProvider`."""

    loki_push_api_alert_rules_error = EventSource(LokiPushApiAlertRulesError)
    loki_push_api_endpoint_departed = EventSource(LokiPushApiEndpointDeparted)
    loki_push_api_endpoint_joined = EventSource(LokiPushApiEndpointJoined)
``` 

Will create an issue to address it later.

Reply:
:+1: to this, if only because where it's coming from is already pretty explicit from the library name/instantiated classes.